.github/workflows/test-and-check.yml

```diff
@@ -1,10 +1,12 @@
 name: Tests + Checks
 
-on: pull_request
+on:
+  pull_request:
+    types: [opened, synchronize, reopened, labeled, unlabeled]
 
 jobs:
   add-to-project:
-    if: github.ref != 'refs/heads/changeset-release/main'
+    if: github.head_ref != 'changeset-release/main'
```

Comment:
Turns out `github.ref` for `pull_request` events refers to the pull request ref (e.g. `refs/6770/merge`)